### PR TITLE
Use nodejs version plugin

### PR DESCRIPTION
**Pull request by blink1073**
cc @fcollonval @agoose77 @ofek

### CODE REVIEW

**Comment by blink1073** [comment]
@fcollonval this approach won't work unless we backport the changes to `federated_labextensions.py` to look for the package name in `pyproject.toml` [first](https://github.com/jupyterlab/jupyterlab/blob/1a07f39bb530a29f897d9127a514630bf52cc914/jupyterlab/federated_labextensions.py#L432).

**Comment by ofek** [comment]
lgtm. here's another example from earlier today https://github.com/squidfunk/mkdocs-material/blob/9603c71adafe5959e57584a4982ee03d340a7be3/pyproject.toml

**Comment by agoose77** [comment]
@blink1073 I'm not sure I'm following. Do you mean for older JupyterLab versions that won't have the latest `federated_labextensions.py`?

**Comment by agoose77** [comment]
I realised that the docs weren't clear for the metadata hook, so I've updated them. The `fields` setting corresponds to the names in the `pyproject.toml` file that are provided by the extension. This is so that other metadata plugins can be used alongside this extension (as hatch-nodejs-version does more than one thing).

File: {{cookiecutter.python_name}}/pyproject.toml [hunk 1]
```diff
@@ -40,8 +36,11 @@ test = [{% if cookiecutter.test.lower().startswith('y') %}
     "pytest-tornasync"{% endif %}
 ]
 
-[project.urls]
-Homepage = "{{ cookiecutter.repository }}"
+[tool.hatch.version]
+source = "nodejs"
+
+[tool.hatch.metadata.hooks.nodejs]
+fields = ["description", "author", "homepage", "repository"]
```
Comment (agoose77) [inline {{cookiecutter.python_name}}/pyproject.toml]:
```suggestion
fields = ["description", "authors", "urls"]
```

Comment (agoose77) [reply {{cookiecutter.python_name}}/pyproject.toml]: Do you want `keywords` too?

**Comment by blink1073** [comment]
> Do you mean for older JupyterLab versions that won't have the latest federated_labextensions.py?

Yes, the changes are in 4.0alpha but have not been backported to 3.x yet.

**Comment by blink1073** [comment]
https://github.com/jupyterlab/jupyterlab/pull/13076

File: {{cookiecutter.python_name}}/pyproject.toml [hunk 2]
```diff
@@ -30,6 +25,7 @@ classifiers = [
 dependencies = [{% if cookiecutter.kind.lower() == "server" %}
     "jupyter_server>=1.6,<2"{% endif %}
 ]
+dynamic = ["version", "description", "authors", "urls"]
```
Comment (blink1073) [inline {{cookiecutter.python_name}}/pyproject.toml]:
```suggestion
dynamic = ["version", "description", "authors", "urls", "keywords"]
```

File: {{cookiecutter.python_name}}/pyproject.toml [hunk 3]
```diff
@@ -1,18 +1,12 @@
 [build-system]
-requires = ["hatchling>=1.3.1", "jupyterlab~=3.1"]
+requires = ["hatchling>=1.3.1", "jupyterlab~=3.1", "hatch-nodejs-version"]
```
Comment (fcollonval) [inline {{cookiecutter.python_name}}/pyproject.toml]: Ensure jupyterlab 3.4.7 or later is used

```suggestion
requires = ["hatchling>=1.3.1", "jupyterlab>=3.4.7,<4.0.0", "hatch-nodejs-version"]
```

**Pull request merged**